test/app_test.py

```diff
@@ -391,8 +392,49 @@ async def app_logs_pty(servicer, stream):
     with servicer.intercept() as ctx:
         ctx.set_responder("AppGetLogs", app_logs_pty)
 
-        with app.run(client=client):
-            time.sleep(0.1)
+        with OutputManager.enable_output():
+            with app.run(client=client):
+                time.sleep(0.1)
 
     captured = capsys.readouterr()
     assert captured.out.endswith("\nsome data\n\r")
+
+
+def test_show_progress_deprecations(client, monkeypatch):
+    # Unset env used to disable warning
+    monkeypatch.delenv("MODAL_DISABLE_APP_RUN_OUTPUT_WARNING")
+
+    app = App()
+
+    # If show_progress is not provided, and output is not enabled, warn
+    with pytest.warns(DeprecationError, match="enable_output"):
+        with app.run(client=client):
+            pass
+
+    # If show_progress is not provided, and output is enabled, no warning
+    with enable_output():
+        with app.run(client=client):
+            pass
+
+    # If show_progress is set to True, and output is not enabled, warn
+    # This is the only one that I'm not sure about - should we auto-enable output for now?
```

Comment:
I think this should show output during the deprecation, yeah.

Reply:
Yeah — it occurred to me that one reason you might (a) be running programmatically and (b) care about the output is if you're running tests. I'd be peeved if my CI broke unexpectedly on a client version update.